feat: Design System Publisher documentation (#394)
vaadin/docs: articles/tools/dspublisher/custom-theme.asciidoc

---
title: Custom Theme
order: 60
---

> = Applying a Custom Design System Theme

Author: Maybe we reserve the term “theme” just for the Vaadin theme, and use some other terminology for the docs website “styling”?

> = Applying a Custom Design System Theme

Reviewer: That's actually a good idea! Needs changes in multiple places unfortunately.

> = Applying a Custom Design System Theme

Reviewer: I would still rename this to "application theme"

> = Applying a Custom Design System Theme

Author: ```suggestion
= Applying a Custom Application Theme
```

> A design system theme is used from a Maven repository, such as Maven Central or a local repository.

Reviewer: Would still call this "application theme"

> A design system theme is used from a Maven repository, such as Maven Central or a local repository.

Author: ```suggestion
An application theme is used from a Maven repository, such as Maven Central or a local repository.
```

Typically the JAR is then installed into a Maven repository like Maven Central or a local repository.

The dependency is then added to the [filename]#pom.xml# in the project's root folder:

.[filename]#pom.xml#
[source,xml]
----
<dependencies>
  ...
  <dependency>
    <groupId>com.example</groupId>
    <artifactId>exampletheme</artifactId>
    <version>1.0</version>
  </dependency>
</dependencies>
----

Next, the theme needs to be applied to the component samples.
Change the <<{articles}/ds/customization/custom-theme/#applying-a-custom-theme, `@Theme` annotation>> in `src/main/java/com/vaadin/demo/DemoExporter.java`:

.[filename]#src/main/java/com/vaadin/demo/DemoExporter.java#
[source,java]
----
@Theme("exampletheme")
public abstract class DemoExporter<T extends Component> extends WebComponentExporter<T> {
  ...
}
----

The website needs to be rebuilt, or restarted in development mode, in order for this change to take effect.

> .Application Theme vs Website Theme

Author: Oh, yes, that’s what we’ve used in other places.

> .Application Theme vs Website Theme

Author: ```suggestion
.Application theme vs website theme
```
